File: backend/app/schemas/content.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ContentCreate(BaseModel):
    title: str = Field(..., min_length=1, max_length=500)
    body: str | None = None
    content_type: str = Field(default="news", pattern="^(news|training|safety|survey)$")
    media_url: str | None = Field(default=None, max_length=1000)
    target_sites: list[str] | None = None
    target_departments: list[str] | None = None
    target_shifts: list[str] | None = None
    expires_at: datetime | None = None


class ContentUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=500)
    body: str | None = None
    content_type: str | None = Field(default=None, pattern="^(news|training|safety|survey)$")
    media_url: str | None = None
    target_sites: list[str] | None = None
    target_departments: list[str] | None = None
    target_shifts: list[str] | None = None
    expires_at: datetime | None = None


class ContentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    tenant_id: uuid.UUID
    title: str
    body: str | None = None
    content_type: str
    media_url: str | None = None
    target_sites: list | None = None
    target_departments: list | None = None
    target_shifts: list | None = None
    published_at: datetime | None = None
    expires_at: datetime | None = None
    created_by: uuid.UUID | None = None
    is_active: bool
    created_at: datetime
    updated_at: datetime


class ContentListResponse(BaseModel):
    data: list[ContentResponse]
    total: int
    page: int = 1
    pages: int = 1

File: frontend/src/types/content.ts
export type ContentType = 'news' | 'training' | 'safety' | 'survey';

export type ContentStatus = 'draft' | 'published' | 'archived';

export interface Content {
  id: string;
  tenant_id: string;
  title: string;
  body: string | null;
  content_type: ContentType;
  media_url: string | null;
  target_sites: string[] | null;
  target_departments: string[] | null;
  target_shifts: string[] | null;
  published_at: string | null;
  expires_at: string | null;
  created_by: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface ContentCreate {
  title: string;
  body?: string;
  content_type?: ContentType;
  media_url?: string;
  target_sites?: string[];
  target_departments?: string[];
  target_shifts?: string[];
  expires_at?: string;
}

export interface ContentUpdate {
  title?: string;
  body?: string;
  content_type?: ContentType;
  media_url?: string;
  target_sites?: string[];
  target_departments?: string[];
  target_shifts?: string[];
  expires_at?: string;
}

export interface ContentListParams {
  content_type?: ContentType;
  is_active?: boolean;
  site_id?: string;
  page?: number;
  page_size?: number;
}

export interface ContentListResponse {
  data: Content[];
  total: number;
  page: number;
  pages: number;
}

export function deriveStatus(content: Content): ContentStatus {
  if (!content.is_active) return 'archived';
  if (content.published_at) return 'published';
  return 'draft';
}

export const CONTENT_TYPE_LABELS: Record<ContentType, string> = {
  news: 'Actualité',
  training: 'Formation',
  safety: 'Sécurité',
  survey: 'Sondage',
};

export const CONTENT_TYPE_ICONS: Record<ContentType, string> = {
  news: 'newspaper',
  training: 'school',
  safety: 'shield',
  survey: 'poll',
};
